### PR TITLE
rPackages.opencv: fixed build

### PR DESCRIPTION
**Pull request by b-rodrigues**
## Description of changes

Fixed build and tested basic functionality.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by Kupac — CHANGES_REQUESTED** [review]
It builds, but something is still missing:
```
ocv_display(faces)
Error: OpenCV(4.9.0) /build/source/modules/highgui/src/window.cpp:1255: error: (-2:Unspecified error) The function is not implemented. Rebuild the library with Windows, GTK+ 2.x or Cocoa support. If you are on Ubuntu or Debian, install libgtk2.0-dev and pkg-config, then re-run cmake or configure script in function 'cvNamedWindow'
```

In addition, native and buildInputs should be split correctly.

**Comment by b-rodrigues** [comment]
I've added `pkgs.gtk2-x11.dev` but the same error still pops up. Any ideas?

File: pkgs/development/r-modules/default.nix [hunk 1]
```diff
@@ -571,6 +572,7 @@ let
     rsvg = [ pkgs.librsvg.dev ];
     ssh = with pkgs; [ libssh ];
     s2 = [ pkgs.openssl.dev ];
+    opencv = [ pkgs.opencv pkgs.gtk2-x11.dev ];
```
Comment (Kupac) [inline pkgs/development/r-modules/default.nix:575]:
```suggestion
    opencv = [ pkgs.opencv ];
```
gtk2 didn't help, so let's remove it. I think we can live with this error for now. I tried a few things, but didn't help either.

**Review by Kupac — CHANGES_REQUESTED** [review]

**Comment by b-rodrigues** [comment]
Ok, think I got it, the error message was related to opencv, not the R package. So I needed to override opencv to build with gtk2 support. That seems to have done the trick, could you test it on your end as well?

**Comment by Kupac** [comment]
OK, I'm building it on x86_64 Linux. If someone could check on darwin, that would be awesome. Not sure they have gtk there? Does it need Cocoa or something?

**Review by Kupac — APPROVED** [review]
Nice detective work! Compiles and works for me and displays images too

image

**Comment by philipp-baumann** [comment]
Confirm it compiles and is functional on `aarch64-darwin`

Screenshot 2024-03-11 at 22 11 31

**Pull request merged**